crates/astria-sequencer/src/sequencer.rs

```diff
@@ -90,22 +104,46 @@ impl Sequencer {
             .finish()
             .ok_or_else(|| anyhow!("server builder didn't return server; are all fields set?"))?;
 
+        let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+
         let grpc_addr = config
             .grpc_addr
             .parse()
             .context("failed to parse grpc_addr address")?;
-        start_grpc_server(&storage, grpc_addr)?;
+        start_grpc_server(&storage, grpc_addr, shutdown_rx)?;
 
         info!(config.listen_addr, "starting sequencer");
-        server
-            .listen_tcp(&config.listen_addr)
-            .await
-            .expect("should listen");
+        select! {
+            _ = signals.stop_rx.changed() => {
+                info!("shutting down sequencer");
+            }
+
+            res = server.listen_tcp(&config.listen_addr) => {
+                match res {
+                    Ok(()) => {
+                        // this shouldn't happen, as there isn't a way for the ABCI server to exit
+                        info!("server exited successfully");
+                    }
+                    Err(e) => {
+                        error!(?e, "server exited with error");
```

Comment:
mention that the abci server exited. We have 2 servers now ;)

Also, no debug fields: `e as &dyn std::error::Error` or `as_ref()` depending on what this thing is